haystack/nodes/prompt/prompt_node.py

```diff
@@ -251,26 +228,24 @@ def create_invocation_layer(self) -> PromptModelInvocationLayer:
         }
         all_kwargs = {**self.model_kwargs, **kwargs}
 
+        if invocation_layer_class:
+            return invocation_layer_class(
+                model_name_or_path=self.model_name_or_path, max_length=self.max_length, **all_kwargs
+            )
         # search all invocation layer classes and find the first one that supports the model,
         # then create an instance of that invocation layer
-        for invocation_layer in self.invocation_layer_classes:
+        for invocation_layer in PromptModelInvocationLayer.invocation_layer_providers:
             if invocation_layer.supports(self.model_name_or_path, **all_kwargs):
                 return invocation_layer(
                     model_name_or_path=self.model_name_or_path, max_length=self.max_length, **all_kwargs
                 )
         raise ValueError(
             f"Model {self.model_name_or_path} is not supported - no matching invocation layer found."
-            f" Currently supported invocation layers are: {self.invocation_layer_classes}"
-            f" You can implement and provide custom invocation layer for {self.model_name_or_path} via PromptModel init."
+            f" Currently supported invocation layers are: {PromptModelInvocationLayer.invocation_layer_providers}"
+            f" You can implement and provide custom invocation layer for {self.model_name_or_path} by subclassing "
+            "PromptModelInvocationLayer."
         )
 
-    def register(self, invocation_layer: Type[PromptModelInvocationLayer]):
-        """
-        Registers additional prompt model invocation layer. It takes a function that returns a boolean as a
-        matching condition on `model_name_or_path` and a class that implements `PromptModelInvocationLayer` interface.
-        """
-        self.invocation_layer_classes.append(invocation_layer)
-
```

Comment:
No need for this function anymore: all subclasses of `PromptModelInvocationLayer` are automatically registered as soon as they're imported / defined.